cmake/compiler_config.cmake

```diff
@@ -102,6 +102,51 @@ set(MSVC_DEFINITIONS
     # Needed to prevent Python from adding a define for snprintf
     # since it was added in Visual Studio 2015.
     HAVE_SNPRINTF
+
+    # In debug builds, Boost's wrap_python.hpp will #undef _DEBUG when BOOST_DEBUG_PYTHON is not defined,
+    # and will then include pyconfig.h :
+    # https://github.com/boostorg/python/blob/boost-1.81.0/include/boost/python/detail/wrap_python.hpp#L23-L57
+    # pyconfig.h will then autolink the Python lib; however, since _DEBUG was #undef'd, it will try to
+    # link against the release version of the Python lib, causing a linker error :
+    # https://github.com/python/cpython/blob/v3.11.4/PC/pyconfig.h#L269-L286
+    $<$<STREQUAL:${CMAKE_BUILD_TYPE},Debug>:BOOST_DEBUG_PYTHON>
```

Comment:
Are you sure this works? Generator expressions are only known at build stage. I don't think it works in a cmake `set()`

Reply:
The generator expression just gets stored as-is in the variable, and since we only use the variable in `target_compile_options`, it ends up spitting the generator expression back out, so it's pretty much equivalent. This could cause issues if we start trying to use the variable in pre-build stages and expecting it to be evaluated, but that seems somewhat unlikely to me. We have a couple generator expressions in MSVC_FLAGS already too (e.g. using `BUILD_STRICT_MODE` to treat warnings as errors). And a good indication that it works is that debug builds should have failed otherwise (unless the issue is no longer there, which I doubt).

Reply:
Ah yes sorry I missed the fact that we already have generator expressions in these cmake `set()`. I'm okay with this change then.